client/state/data-layer/wpcom/sites/site-user-type/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_CONNECT_SAVE_SITE_USER_TYPE } from 'state/action-types';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+/**
+ * Save the site's user type.
+ *
+ * @param   {object} action Save site's user type action.
+ * @returns {object}        The dispatched action.
+ */
+export const saveSiteUserType = action =>
+	http(
+		{
+			method: 'POST',
+			apiNamespace: 'wpcom/v2',
+			path: `/sites/${ action.siteId }/site-ownership`,
```

Comment:
This doesn't seem to match the current endpoint route, which is `site-owner`. 

Also, I've suggested some changes on that in the diff - so we might want to address them first.